services/blockstorage/service.go

```diff
@@ -62,12 +63,91 @@ func (s *service) CommitBlock(input *services.CommitBlockInput) (*services.Commi
 	return nil, nil
 }
 
+func (s *service) lookForTransactionsBlockHeader(height primitives.BlockHeight) (*services.GetTransactionsBlockHeaderOutput, error) {
+	txBlock, err := s.persistence.GetTransactionsBlock(height)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &services.GetTransactionsBlockHeaderOutput{
+		TransactionsBlockProof: txBlock.BlockProof,
+		TransactionsBlockHeader: txBlock.Header,
+		TransactionsBlockMetadata: txBlock.Metadata,
+	}, nil
+}
+
 func (s *service) GetTransactionsBlockHeader(input *services.GetTransactionsBlockHeaderInput) (*services.GetTransactionsBlockHeaderOutput, error) {
```

Comment:
a method that opens a long-lived goroutine might need to take a `Context` argument to support cancellation (for instance, on shutdown / test end). See PR #29. This would mean changing the API, but I think this might be required anyway. @talkol @OdedWx thoughts?

Reply:
I have introduced the config to fix this.